### DISCUSSION TITLE
"underlying network changed", `Web3Provider(provider, 'any')`

### DISCUSSION THREAD

**Discussion by ackvf** [wevm/wagmi#1277] [Q&A]
I am receiving this error, after to me yet unknown circumstances.
![image](https://user-images.githubusercontent.com/2313018/200984441-dcf27d77-6e20-4ad1-9e0e-a07c006b89d2.png)

I have learned that it is possible to work around this using this solution https://github.com/Uniswap/web3-react/issues/127

```
const getLibrary = (provider, connector) => {
	return new ethers.providers.Web3Provider(provider, 'any') // instead of Web3Provider(provider)
};
```

But how do I manage that with wagmi, where everything is abstracted away?

**Comment by 0xxmemo** [comment]
Im having the same problem can someone help please

**Comment by josh-richardson** [comment]
I'm also encountering this issue.

**Comment by 0xxmemo** [comment]
Okay I fixed the issue;

Im using Next.js with TypeScript;

I created a type file and declared Avalanche > this is how I did it >

src/types/wagmi.d.ts

```
import { Chain } from "wagmi";

declare const chainId: {
    readonly avalanche: 43_114;
}
declare const avalanche: Chain;
```

You can add whaterver other network here and it should work ( just be sure the name & id in your custom chains config is the same with this declaration )

tsconfig.json

```
{
  "compilerOptions": {
...
  "include": [
    "next-env.d.ts",
    "types/*.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

**Reply by 0xxmemo** [reply]
This is actually fixed on "ethers": "^5.7.2" & latest 2 wagmi versions!

**Reply by Nikitulb** [reply]
Hello. Are you just only add .d.td file and in works ? What's about this solution new ethers.providers.Web3Provider(provider, 'any')?
